### PR TITLE
[SPARK-25170][DOC] Add list and short description of Spark Executor Task Metrics to the documentation

### PR DESCRIPTION
**Pull request by LucaCanali**
## What changes were proposed in this pull request?

Add description of Task Metrics to the documentation.

## How was this patch tested?

None.

### CODE REVIEW

File: docs/monitoring.md [hunk 1]
```diff
@@ -388,6 +388,163 @@ value triggering garbage collection on jobs, and `spark.ui.retainedStages` that
...
+task execution level. The metrics can be used for performance troubleshooting.
+A list of the available metrics with a short description:
+
+<table class="table">
+  <tr><th>Spark Executor Task Metric name</th>
+      <th>Short description</th>
+  </tr>
+  <tr>
+    <td>executorRunTime</td>
+    <td>Time the executor spent running this task. This includes time fetching shuffle data.
```
Comment (kiszk) [inline docs/monitoring.md]: Does `Time` mean `elapsed time` or other `time`?

File: docs/monitoring.md [hunk 2]
```diff
@@ -388,6 +388,163 @@ value triggering garbage collection on jobs, and `spark.ui.retainedStages` that
...
+    The value is expressed in milliseconds.</td>
+  </tr>
+  <tr>
+    <td>executorCpuTime
```
Comment (kiszk) [inline docs/monitoring.md]: Do we miss `<td>` in these two lines?

File: docs/monitoring.md [hunk 3]
```diff
@@ -388,6 +388,163 @@ value triggering garbage collection on jobs, and `spark.ui.retainedStages` that
...
+    <td>CPU Time taken on the executor to deserialize this task.
+     The value is expressed in nanoseconds.</td>
+  </tr>
+  <tr>
+    <td>resultSize</td>
+    <td>The number of bytes this task transmitted back to the driver as the TaskResult.</td>
+  </tr>
+  <tr>
+    <td>jvmGCTime</td>
+    <td>Amount of time the JVM spent in garbage collection while executing this task.
```
Comment (kiszk) [inline docs/monitoring.md]: Why do we start with `amount of` while the above parameters start `Time` or `CPU Time`?

File: docs/monitoring.md [hunk 4]
```diff
@@ -388,6 +388,163 @@ value triggering garbage collection on jobs, and `spark.ui.retainedStages` that
...
+    The value is expressed in milliseconds.</td>
+  </tr>
+  <tr>
+    <td>resultSerializationTime</td>
+    <td>Amount of time spent serializing the task result.
```
Comment (kiszk) [inline docs/monitoring.md]: ditto

**Comment by kiszk** [comment]
I like to add description for metrics.

**Comment by LucaCanali** [comment]
Thanks @kiszk for reviewing this. I have addressed your comments in a new commit +
apologies as I have now moved this to a new PR https://github.com/apache/spark/pull/22397 
I am closing this to avoid confusion.

**Pull request closed without merging**